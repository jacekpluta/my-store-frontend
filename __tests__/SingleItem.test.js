import React from "react";
import SingleItem, { SINGLE_ITEM_QUERY } from "../components/SingleItem";
import { mount } from "enzyme";
import { MockedProvider } from "@apollo/react-testing";

const fakeItem = {
  id: "abc123",
  title: "test",
  description: "test description",
  largeImage: "test.jpg",
};
it("renders correctly", async () => {
  const mocks = [
    {
      request: { query: SINGLE_ITEM_QUERY, variables: { id: "123" } },
      result: {
        data: {
          item: fakeItem,
        },
      },
    },
  ];

  const wrapper = mount(
    <MockedProvider mocks={mocks} addTypename={false}>
      <SingleItem itemId="123" />
    </MockedProvider>
  );
  expect(wrapper.text()).toContain("Loading...");

  await new Promise((resolve) => setTimeout(resolve, 0));

  wrapper.update();

  expect(wrapper.find("h2")).toMatchSnapshot();
  expect(wrapper.find("img")).toMatchSnapshot();
  expect(wrapper.find("p")).toMatchSnapshot();
});

it("errors when no item found", async () => {
  const mocks = [
    {
      request: { query: SINGLE_ITEM_QUERY, variables: { id: "123" } },
      result: {
        errors: [{ message: "Item not found" }],
      },
    },
  ];

  const wrapper = mount(
    <MockedProvider mocks={mocks} addTypename={false}>
      <SingleItem itemId="123" />
    </MockedProvider>
  );

  await new Promise((resolve) => setTimeout(resolve, 0));

  wrapper.update();

  const item = wrapper.find('[data-test="graphql-error"]');
  expect(item.text()).toContain("Error!Item not found");
  expect(item).toMatchSnapshot();
});
