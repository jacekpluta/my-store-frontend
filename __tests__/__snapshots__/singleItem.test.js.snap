// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errors when no item found 1`] = `
<p
  data-test="graphql-error"
>
  <strong>
    Error!
  </strong>
  Item not found
</p>
`;

exports[`renders correctly 1`] = `null`;

exports[`renders correctly 2`] = `null`;

exports[`renders correctly 3`] = `
<p>
  Loading...
</p>
`;
